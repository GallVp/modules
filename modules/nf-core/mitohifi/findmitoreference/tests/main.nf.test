
nextflow_process {

    name "Test Process MITOHIFI_FINDMITOREFERENCE"
    script "../main.nf"
    process "MITOHIFI_FINDMITOREFERENCE"

    tag "modules"
    tag "modules_nfcore"
    tag "mitohifi"
    tag "mitohifi/findmitoreference"

    test("test-mitohifi-findmitoreference") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test' ],       // meta map
				    "Phalera flavescens" // species
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-mitohifi-findmitoreference-stub") {
        options '-stub'
        when {
            process {
                """
                input[0] = [
				    [ id:'test' ],       // meta map
				    "Phalera flavescens" // species
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
