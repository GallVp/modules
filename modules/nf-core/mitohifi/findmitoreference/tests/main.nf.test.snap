{
    "test-mitohifi-findmitoreference-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "accession.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "accession.gb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,ca8d8bb4eccd4fec95ac8cf7f45c3328"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "accession.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gb": [
                    [
                        {
                            "id": "test"
                        },
                        "accession.gb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ca8d8bb4eccd4fec95ac8cf7f45c3328"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T13:49:13.884847"
    },
    "test-mitohifi-findmitoreference": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "NC_016067.1.fasta:md5,ca39f522220f12582245bbbbb8e79fbc"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "NC_016067.1.gb:md5,d73f4d89bdedeb51c0d657632d577b26"
                    ]
                ],
                "2": [
                    "versions.yml:md5,ca8d8bb4eccd4fec95ac8cf7f45c3328"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "NC_016067.1.fasta:md5,ca39f522220f12582245bbbbb8e79fbc"
                    ]
                ],
                "gb": [
                    [
                        {
                            "id": "test"
                        },
                        "NC_016067.1.gb:md5,d73f4d89bdedeb51c0d657632d577b26"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ca8d8bb4eccd4fec95ac8cf7f45c3328"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T13:49:08.956252"
    }
}